### PR TITLE
LPS-171144 Implement backend of the new configuration fields

### PR DESCRIPTION
**Pull request by lfbesada**
https://issues.liferay.com/browse/LPS-171144

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by lfbesada** [comment]
ci:test:sf

**Comment by lfbesada** [comment]
ci:test:relevant

**Comment by lfbesada** [comment]
ci:test:relevant

**Comment by ealonso** [comment]
Just started reviewing :)

File: modules/apps/asset/asset-categories-admin-web/src/main/java/com/liferay/asset/categories/admin/web/internal/info/collection/provider/AssetEntriesWithSameAssetCategoryRelatedInfoItemCollectionProvider.java [hunk 1]
```diff
@@ -138,6 +151,52 @@ public Class<?> getSourceItemClass() {
...
+			SelectInfoFieldType.OPTIONS,
+			ListUtil.fromArray(
+				new SelectInfoFieldType.Option(
+					new ResourceBundleInfoLocalizedValue(
+						getClass(), "not-selected"),
+					StringPool.BLANK),
+				new SelectInfoFieldType.Option(
+					new ResourceBundleInfoLocalizedValue(
+						getClass(), "any-category-of-the-same-vocabulary"),
+					"anyAssetCategory"),
```
Comment (ruben-pulido) [inline modules/apps/asset/asset-categories-admin-web/src/main/java/com/liferay/asset/categories/admin/web/internal/info/collection/provider/AssetEntriesWithSameAssetCategoryRelatedInfoItemCollectionProvider.java:172]: I would rename the key to "anyCategoryOfTheSameVocabulary" to be more descriptive

**Comment by ealonso** [comment]
see: https://github.com/liferay-echo/liferay-portal/pull/10656

**Pull request closed without merging**